docs/AQL/master03-syntax.adoc

```diff
@@ -512,42 +517,169 @@ FROM
 
 ==== COUNT
 
-The `COUNT()` function returns the number of values of given _expression_ argument. The syntax is `COUNT([DISTINCT] expression |\*)`.
+The `COUNT` function returns the number of values of given _expression_ argument. The syntax is `COUNT([DISTINCT] expression |\*)`.
 
 The `COUNT(*)` is used to calculate the number of rows of the result set, whether or not they contain `NULL`.
-The `DISTINCT` keyword can be calculate the number of only the distinct values of _expression_.
+The `DISTINCT` keyword can be used to calculate the number of only distinct values of _expression_.
 
-If there are no matching rows, `COUNT()` returns `0`.
+If there are no matching rows, then this function returns `0`.
 Input values type can be anything, while the return type is always an Integer.
 
 ==== MIN
 
-The `MIN()` function returns the minimum value of given _expression_ argument. The syntax is `MIN(expression)`.
+The `MIN` function returns the minimum value of given _expression_ argument. The syntax is `MIN(expression)`.
 
-If there are no matching rows, `MIN()` returns `NULL`.
+If there are no matching rows, then this function returns `NULL`.
```

Comment:
Normally functions like Min(), Max() etc return numbers, either Integer, Real or Numeric, depending on the type system. I thought that in SQL they returned the minimum numeric value from a list (or set) of numeric values as per here: https://specifications.openehr.org/releases/BASE/latest/base_types.html#_builtins_package . If the container is empty, I would expect min(), max(), avg() etc to all return 0. If the argument is an expression, then it has to be one that evaluates to a Set<Numeric> or similar. I cannot find online however what these functions in SQL return if they are passed an empty column.

This also applies to most/all the functions below I guess.

Reply:
Well I actually checked at that time I wrote MySQL, Postgres, Oracle, Firebird and SQL server - they all return NULL.
I think the logic is that if the list/set is empty, then you cannot give any number as min/max/avg/sum result; it is a sort of an exceptional situation. If we return 0 in such situation, I suggest we also change that to be consistent.

Reply:
For some implem languages that would be pretty strange that a function could return a primitive value (Integer, Numeric or whatever) or a Null - that means the formal definition of the functions are that they return full object types, not value types. Which I guess is fine, but it's not really consistent with the more primitive functions we currently have defined. I'd be interested to know what Better (or anyone) has implemented - because if a function like Min() could return Null, and you use it in any kind of expression in AQL, you have to wrap it in the usual null-checking code to avoid an exception. Or - do we limit it to being used only in the FROM part?